### PR TITLE
Fix/teams compliance recording policy

### LINKED ISSUE #3712: MSFT_TeamsComplianceRecordingPolicy attribute not exported correctly
### Description of the issue

Export of this resource looks like this:
TeamsComplianceRecordingPolicy "TeamsComplianceRecordingPolicy-Tag:CompRec"
        {
            ApplicationId                                       = $ConfigurationData.NonNodeData.ApplicationId;
            CertificateThumbprint                               = $ConfigurationData.NonNodeData.CertificateThumbprint;
            ComplianceRecordingApplications                     = @("Microsoft.Teams.Policy.Administration.Cmdlets.Core.ComplianceRecordingApplication");
            Description                                         = "Managed by Contoso";
            DisableComplianceRecordingAudioNotificationForCalls = $True;
            Enabled                                             = $True;
            Ensure                                              = "Present";
            Identity                                            = "Tag:CompRec";
            TenantId                                            = $OrganizationName;
            WarnUserOnRemoval                                   = $True;
        }
The Attribute "ComplianceRecordingApplications" contains the Object Name istead of the Value

### Microsoft 365 DSC Version

V1.23.809.1

### Which workloads are affected

Teams

### The DSC configuration

_No response_

### Verbose logs showing the problem

_No response_

### Environment Information + PowerShell Version

_No response_

### PR DESCRIPTION
**Pull request by sandrola**
#### Pull Request (PR) description
Add CIM Instance to  MSFT_TeamsComplianceRecordingPolicy to be able to compare with configured policy

#### This Pull Request (PR) fixes the following issues
- Fixes #3712

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 # UNRELEASED
 
+* TeamsComplianceREcordingPolicy
```
Comment (ykuijs) [inline CHANGELOG.md:5]: Could you update your branch with Dev and then place this item alphabetically on the right place? Just the MISC and DEPENDENCIES ones should be at the bottom, the rest alphabetically.

File: Modules/Microsoft365DSC/DSCResources/MSFT_TeamsComplianceRecordingPolicy/MSFT_TeamsComplianceRecordingPolicy.psm1 [hunk 2]
```diff
@@ -190,13 +214,55 @@ function Set-TargetResource
...
+                $keyName = $key.substring(0, 1).ToLower() + $key.substring(1, $key.length - 1)
                 $keyValue = Convert-M365DSCDRGComplexTypeToHashtable -ComplexObject $CreateParameters.$key
                 $CreateParameters.Remove($key) | Out-Null
                 $CreateParameters.Add($keyName, $keyValue)
             }
         }
         Write-Verbose -Message "Creating {$Identity} with Parameters:`r`n$(Convert-M365DscHashtableToString -Hashtable $CreateParameters)"
         New-CsTeamsComplianceRecordingPolicy @CreateParameters | Out-Null
+
+        if($ComplianceRecordingApplications.Count -gt 0){
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/DSCResources/MSFT_TeamsComplianceRecordingPolicy/MSFT_TeamsComplianceRecordingPolicy.psm1:226]: Please adhere to the code styling conventions: Add a space after the keywords (if/else/foreach) and place the opening brace on a new line.

**Review by ykuijs — DISMISSED** [review]
Few small comments

**Comment by andikrueger** [comment]
@sandrola Thanks for your PR and the work you put into it! Any chance you could resolve conflicts and address Yorick's comments?

**Comment by pkernevez** [comment]
As the original issue #3754  has been closed, I think this PR should be close

**Review by andikrueger — COMMENTED** [review]
Just one minor remark and a question for @NikCharlebois

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,8 @@
 * MISC
   * M365DSCReport: Also define property dataType, if present, as being primary key on
     CIM instances.
+* TeamsComplianceREcordingPolicy
```
Comment (andikrueger) [inline CHANGELOG.md:17]: Minor typo REcording

**Review by NikCharlebois — APPROVED** [review]

**Comment by NikCharlebois** [comment]
Can we please merge and update the PR with the latest changes? Thanks

**Comment by FabienTschanz** [comment]
@sandrola Could you please update or close the PR? Thank you.

**Comment by NikCharlebois** [comment]
As we approach our next Major release (Oct 2nd), we need to make confirm this is still a valid PR and update it to resolve any merge conflicts.

**Comment by sandrola** [comment]
The PR is still valid and I just updated it with dev.

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**